### PR TITLE
Update ServiceValidator.java

### PR DESCRIPTION
**Pull request by Codewithsumeet**
Added compile-time validation in the GraphQL compiler plugin to ensure stub entities annotated with @subgraph:Entity`only contain key fields. Non-key fields are now identified and trigger a diagnostic error during validation. The implementation includes extracting key fields from the annotation and validating record fields against these keys.

## Purpose

Fixes: [#4816](https://github.com/ballerina-platform/ballerina-library/issues/4816)

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility
- [ ] No `commons` package changes (if there are any, please update the GraphQL version in [GraphQL tools](https://github.com/ballerina-platform/graphql-tools) and [Ballerina dev tools](https://github.com/ballerina-platform/ballerina-dev-tools))
- [ ] No `compiler` package changes (if there are any, please update the GraphQL version in [Ballerina dev tools](https://github.com/ballerina-platform/ballerina-dev-tools))

### CODE REVIEW

File: compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/service/validator/ServiceValidator.java [hunk 1]
```diff
@@ -185,27 +185,67 @@ private void validateEntities() {
...
+        return; 
+    }
+
+  
+    List<String> keyFields = new ArrayList<>();
+    for (MappingFieldNode fieldNode : entityAnnotation.annotValue().get().fields()) {
+        if (fieldNode.kind() != SPECIFIC_FIELD) {
+            
+            addDiagnostic(CompilationDiagnostic.PROVIDE_KEY_VALUE_PAIR_FOR_ENTITY_ANNOTATION, fieldNode.location());
+            continue;
```
Comment (DimuthuMadushan) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/service/validator/ServiceValidator.java:199]: Invalid indentation. Check other places as well.

File: compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/service/validator/ServiceValidator.java [hunk 2]
```diff
@@ -185,27 +185,67 @@ private void validateEntities() {
...
+                    keyFields.add(keyField.toString().replace("\"", "").trim());
+                }
             }
+        } else if (valueNode.kind() == SyntaxKind.STRING_LITERAL) {
+            
+            keyFields.add(valueNode.toString().replace("\"", "").trim());
         }
     }
+    return keyFields; 
+}
```
Comment (DimuthuMadushan) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/service/validator/ServiceValidator.java:246]:
```suggestion
        List<String> keyFields = new ArrayList<>();
        if (specificFieldNode.valueExpr().isPresent()) {
            ExpressionNode valueNode = specificFieldNode.valueExpr().get();
            if (valueNode.kind() == SyntaxKind.LIST_CONSTRUCTOR) {
                for (Node keyField : ((ListConstructorExpressionNode) valueNode).expressions()) {
                    if (keyField.kind() == SyntaxKind.STRING_LITERAL) {
                        keyFields.add(keyField.toString().replace("\"", "").trim());
                    }
                }
            } else if (valueNode.kind() == SyntaxKind.STRING_LITERAL) {
                keyFields.add(valueNode.toString().replace("\"", "").trim());
            }
        }
        return keyFields; 
    }
```

**Comment by DimuthuMadushan** [comment]
hi @Codewithsumeet ! are you still working on this?

**Comment by Codewithsumeet** [comment]
Sorry, im not working on this

**Pull request closed without merging**